MekHQ/src/mekhq/campaign/personnel/ranks/RankSystem.java

```diff
@@ -0,0 +1,320 @@
+import mekhq.campaign.io.Migration.PersonMigrator;
+import mekhq.campaign.personnel.enums.RankSystemType;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ResourceBundle;
+import java.util.stream.Stream;
+
+public class RankSystem implements Serializable {
+    //region Variable Declarations
+    private static final long serialVersionUID = -6037712487121208137L;
+
+    private String code; // Primary Key, must be unique
+    private String name;
+    private String description;
+    private transient RankSystemType type; // no need to serialize
+    private List<Rank> ranks;
+    //endregion Variable Declarations
+
+    //region Constructors
+    private RankSystem(final RankSystemType type) {
+        this("UNK", "Unknown", "", type);
+    }
+
+    public RankSystem(final RankSystem rankSystem) {
+        setCode(rankSystem.getCode());
+        setName(rankSystem.toString());
+        setDescription(rankSystem.getDescription());
+        setType(rankSystem.getType());
+        setRanks(new ArrayList<>(rankSystem.getRanks()));
+    }
+
+    public RankSystem(final String code, final String name,
+                      final String description, final RankSystemType type) {
+        setCode(code);
+        setName(name);
+        setDescription(description);
+        setType(type);
+        final RankSystem system = Ranks.getRankSystemFromCode(code);
+        setRanks((system == null) ? new ArrayList<>() : new ArrayList<>(system.getRanks()));
+    }
+    //endregion Constructors
+
+    //region Getters/Setters
+    public String getCode() {
+        return code;
+    }
+
+    public void setCode(final String code) {
+        this.code = code;
+    }
+
+    public void setName(final String name) {
+        this.name = name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(final String description) {
+        this.description = description;
+    }
+
+    public RankSystemType getType() {
+        return type;
+    }
+
+    public void setType(final RankSystemType type) {
+        this.type = type;
+    }
+
+    public List<Rank> getRanks() {
+        return ranks;
+    }
+
+    public void setRanks(final List<Rank> ranks) {
+        this.ranks = ranks;
+    }
+    //endregion Getters/Setters
+
+    //region Boolean Comparison Methods
+    public boolean isWoBMilitia() {
+        return "WOBM".equals(getCode());
+    }
+
+    public boolean isComGuard() {
+        return "CG".equals(getCode());
+    }
+
+    public boolean isCGOrWoBM() {

```

Comment:
Let's document this one.

Reply:
It's used to determine if a person can have a ROM Designation.
Already documented in the Ranks Guide, added a Javadoc comment here, and will open a RFE to make that data-based so it can be used in custom rank systems (for ROM Designation, Manei Domini Class, and Manei Domini Rank)